test_conformance/contractions/contractions.cpp

```diff
@@ -576,56 +576,39 @@ test_status InitCL( cl_device_id device )
         "}\n"
         "\n" };
 
-    for( i = 0; i < sizeof( sizeNames ) / sizeof( sizeNames[0] ); i++ )
+    for (i = 0; i < sizeof(sizeNames) / sizeof(sizeNames[0]); i++)
     {
-        size_t strCount = sizeof( kernels ) / sizeof( kernels[0] );
+        size_t strCount = sizeof(kernels) / sizeof(kernels[0]);
         kernels[0] = "";
 
-        for( j = 2; j < strCount; j += 2 )
-            kernels[j] = sizeNames[i];
-
-        gProgram[i] = clCreateProgramWithSource(gContext, strCount, kernels, NULL, &error);
-        if( NULL == gProgram[i] )
+        for (j = 2; j < strCount; j += 2) kernels[j] = sizeNames[i];
+        const auto error = create_single_kernel_helper(
```

Comment:
Very minor: `error` is already declared on Line 463.  I think it would be better not to add this shadow declaration.  See also Line 604.

```suggestion
        error = create_single_kernel_helper(
```

Reply:
Pushed a new commit with this, was a slight clang-format modification too as a result